### PR TITLE
Use datadog-checks-downloader in the integration command 

### PR DESCRIPTION
**Pull request by zippolyte**
### What does this PR do?

Stop using Datadog/pip with integrated TUF+in-toto, and use the [datadog-checks-downloader](https://github.com/DataDog/integrations-core/tree/master/datadog_checks_downloader) instead

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: omnibus/config/software/datadog-agent-integrations.rb
Comment (trishankatdatadog) [inline omnibus/config/software/datadog-agent-integrations.rb]: let's remove dependency on `datadog-pip` too, pls

File: cmd/agent/app/integrations.go
Comment (trishankatdatadog) [inline cmd/agent/app/integrations.go]:
```suggestion
	tufPkgPattern       = "datadog-.*"
```

Comment (trishankatdatadog) [reply cmd/agent/app/integrations.go]: Because we are not using `pip` to download anymore, we will not get "autocorrection."

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]
Thanks, Hippo --- this looks like a great start!

I have a few comments I'd like to see addressed, please. Also:

* Could we update `tufIndex` to be consistent with https://github.com/DataDog/integrations-core/pull/3087, please?
* Could we get rid of the `search` function, which doesn't work yet anyway?
* When we `pip install` a wheel, are temporarily files / directories created? If so, where?
* Why is it called `getTUFPipCachePath()`?

File: cmd/agent/app/integrations.go
Comment (trishankatdatadog) [inline cmd/agent/app/integrations.go]: Might be a good idea to do an `os.Stat(wheelPath)` here?

File: cmd/agent/app/integrations.go
Comment (trishankatdatadog) [inline cmd/agent/app/integrations.go]: Hmm, not sure this does the right thing. Might be better to do something like `'-' + ('v'*verbose)` in Python?

File: cmd/agent/app/integrations.go
Comment (trishankatdatadog) [inline cmd/agent/app/integrations.go]: Hmm, not sure this does the right thing. Might be better to do something like `'-' + ('v'*verbose)` in Python?

File: cmd/agent/app/integrations.go
Comment (zippolyte) [reply cmd/agent/app/integrations.go]: For the record yeah it does the right thing. But yeah using `strings.Repeat` looks better

**Comment by trishankatdatadog** [comment]
This is really getting into shape --- thanks, @zippolyte!

I have a few requests I hope you can help me with:

1. Could we delete the `/(etc|opt)/datadog-agent/repositories/` _before_ updating to 6.10?
2. Could we delete `/opt/datadog-agent/embedded/lib/pythonX.Y/site-packages/datadog_checks/downloader/data/` when a user _removes_, or _updates_ from, 6.10?

Of course, we would also need to do the same for Windows.

**Comment by trishankatdatadog** [comment]
One more thing: do we want users to be able to install latest versions of integrations (if they don't specify `versionToInstall`)? The downloader will automatically download the latest version if the `--version` is not specified.

**Comment by zippolyte** [comment]
We thought about that at some point I think, and it's not something we wanted to allow, because we still want users to prefer upgrading the agent in most cases. We don't want users to be able to not specify a version, and be able to always upgrade an integration with their config management tool for example.

File: cmd/agent/app/integrations.go
Comment (gzussa) [inline cmd/agent/app/integrations.go]: Let's use `fmt.Sprintf` instead of `+`

File: cmd/agent/app/integrations.go
Comment (gzussa) [inline cmd/agent/app/integrations.go]: ❤️

**Review by gzussa — COMMENTED** [review]
Just a minor comment otherwise looks LGTM 👍 💯

File: cmd/agent/app/integrations.go
Comment (truthbk) [inline cmd/agent/app/integrations.go]: This is a bit of an all or nothing approach with the output, would we get better feedback from the subprocess by doing what we're doing with `stderr` and read from the `downloaderCmd.StdoutPipe()` while executing the command with `downloaderCmd.Start()` `downloaderCmd.Wait()`? 

Of course this all probably only makes sense if we want to get a greater sense of progress from whatever the subprocess is spitting out, stuff like percentage downloaded, validation message, etc. etc (if they exist).

File: cmd/agent/app/integrations.go
Comment (truthbk) [inline cmd/agent/app/integrations.go]: Not sure if we want to support the full version specification defined in [PEP440](https://www.python.org/dev/peps/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions):

Feel free to ignore if our versioning will never contemplate such edge cases.

```
VERSION_PATTERN = r"""
    v?
    (?:
        (?:(?P<epoch>[0-9]+)!)?                           # epoch
        (?P<release>[0-9]+(?:\.[0-9]+)*)                  # release segment
        (?P<pre>                                          # pre-release
            [-_\.]?
            (?P<pre_l>(a|b|c|rc|alpha|beta|pre|preview))
            [-_\.]?
            (?P<pre_n>[0-9]+)?
        )?
        (?P<post>                                         # post release
            (?:-(?P<post_n1>[0-9]+))
            |
            (?:
                [-_\.]?
                (?P<post_l>post|rev|r)
                [-_\.]?
                (?P<post_n2>[0-9]+)?
            )
        )?
        (?P<dev>                                          # dev release
            [-_\.]?
            (?P<dev_l>dev)
            [-_\.]?
            (?P<dev_n>[0-9]+)?
        )?
    )
    (?:\+(?P<local>[a-z0-9]+(?:[-_\.][a-z0-9]+)*))?       # local version
"""
```

They provide a small snippet you could maybe use: 

```python
import re
def is_canonical(version):
    return re.match(r'^([1-9]\d*!)?(0|[1-9]\d*)(\.(0|[1-9]\d*))*((a|b|rc)(0|[1-9]\d*))?(\.post(0|[1-9]\d*))?(\.dev(0|[1-9]\d*))?$', version) is not None
```

File: cmd/agent/app/integrations_nix.go
Comment (truthbk) [inline cmd/agent/app/integrations_nix.go]: This doesn't seem to be used anywhere? 🤔

File: cmd/agent/app/integrations_windows.go
Comment (truthbk) [inline cmd/agent/app/integrations_windows.go]: ditto

File: cmd/agent/app/integrations.go
Comment (truthbk) [inline cmd/agent/app/integrations.go]: looks like we're not calling the downloader executable directly, so the constant we've defined `downloaderBin` I _think_ is useless. Maybe we should call it `downloaderModule = "datadog_checks.downloader"` and use that here.

File: omnibus/resources/agent/msi/source.wxs.erb
Comment (truthbk) [inline omnibus/resources/agent/msi/source.wxs.erb]: We need to check with @derekwbrown if we can just remove this with no side-effects across upgrades, etc. I _think_ it's fine, but let's confirm with him.

**Review by truthbk — COMMENTED** [review]
Nice one! Added some comments, but mostly small stuff, this looks like a huge step in the right direction for the integrations command. 🙇

File: cmd/agent/app/integrations.go
Comment (zippolyte) [reply cmd/agent/app/integrations.go]: they weren't able before anyway, the repo didn't support searching, so using the command just errored. Since the command was hidden anyway, we can safely remove it.

File: cmd/agent/app/integrations.go
Comment (zippolyte) [reply cmd/agent/app/integrations.go]: Yeah I still add the proxy settings for the downloader. And for every pip operation we do, we add `--no-deps` so now that it's installing from an already downloaded wheel, yeah it shouldn't do any network operation

File: cmd/agent/app/integrations.go
Comment (zippolyte) [reply cmd/agent/app/integrations.go]: I think we do. I'll let @trishankatdatadog answer that for more details

File: cmd/agent/app/integrations.go
Comment (zippolyte) [reply cmd/agent/app/integrations.go]: Yeah we have a plan to extend a bit the versioning of our integrations, but for now we are strictly sticking to `major.minor.fix`. I prefer to keep it that way for now, and change it later as we will have to change other things as well for supporting more version formats.

File: cmd/agent/app/integrations_nix.go
Comment (zippolyte) [reply cmd/agent/app/integrations_nix.go]: indeed, forgot to remove that

File: cmd/agent/app/integrations.go
Comment (trishankatdatadog) [reply cmd/agent/app/integrations.go]: I already do this PEP 440 check in my downloader, so we should be good 👍

File: cmd/agent/app/integrations.go
Comment (trishankatdatadog) [reply cmd/agent/app/integrations.go]: We can support it in the future if there is a need for this, yes

**Review by derekwbrown — DISMISSED** [review]
Windows changes LGTM on the surface... I'm assuming someone's done a build and tried the result?

**Comment by trishankatdatadog** [comment]
There is mention of TUF in `--verbose` we should remove, and also `tufPkgPattern`

File: cmd/agent/app/integrations.go
Comment (truthbk) [reply cmd/agent/app/integrations.go]: it would be highly desirable for our customers I think.

File: omnibus/resources/agent/msi/source.wxs.erb
Comment (truthbk) [inline omnibus/resources/agent/msi/source.wxs.erb]: So I just verified this `TufMetaPath` key in the registry corresponds to the `C:\ProgramData\Datadog\repositories\` which is currently removed by the `<util:RemoveFolderEx>` action below. My guess would be that we are already cleaning up after ourselves on uninstall, so I think you'd be good to go on windows. Again, let's have @derekwbrown comment on this. Also, this should be very easy to test during QA.

**Review by truthbk — CHANGES_REQUESTED** [review]
I don't think we're quite there yet on `yum`.

**Review by truthbk — DISMISSED** [review]
I think this looks good now. Make sure you QA all install/uninstall/upgrade flows:
 - install `6.10` on clean box.
 - uninstall `6.10`: should leave no residue.
 - upgrade from `<6.10` (old TUF location should get wiped).
 - upgrade from `6.10.0` to `>6.10.0` (you can use subsequent release candidate to verify this): old metadata should be removed, after every upgrade we should have a clean slate, right?

And also test windows. Please wait for Derek's confirmation on the windows installer before you merge.

**Review by trishankatdatadog — APPROVED** [review]

**Review by truthbk — APPROVED** [review]

**Pull request merged**
